src/features/user/userSlice.ts

```diff
@@ -0,0 +1,50 @@
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+import { RootState } from '../../store';
+import { saveLocalStorage } from '../../utils/localStorage';
+import { fetchLogin, InfoUser } from './userApi';
+export interface UserState {
+  info?: InfoUser;
+  token?: string;
```

Comment:
```suggestion
  info: User | null;
  token: string | null;
```
Should not use undefined in State